src/test.py

```diff
@@ -653,14 +651,14 @@ def setUp(self):
 
     def request(self, url, method='GET', read=True, data={}):
         assert url.startswith('/')
-        wfile = minisix.io.StringIO()
-        rfile = minisix.io.StringIO()
+        wfile = minisix.io.BytesIO()
+        rfile = minisix.io.BytesIO()
         connection = FakeHTTPConnection(wfile, rfile)
         connection.putrequest(method, url)
         connection.endheaders()
         rfile.seek(0)
-        wfile.seek(0)
         handler = TestRequestHandler(rfile, wfile)
+        wfile.seek(0)
```

Comment:
wow how did that even work before that commit

Reply:
I don't think it ever did, these classes just aren't used.  All the code that would be required to write testcases with them is there, but as far as I can tell, they never ended up being used in actual test cases.

Reply:
They are used in the WebStats plugin in https://github.com/ProgVal/Supybot-plugins/ (although its tests are disabled on py3 because I was too lazy to look into them)

Reply:
hmm but you're right, those tests are really basic, so they just happened to work.